cloud/openstack/os_api_endpoint.py

```diff
@@ -0,0 +1,591 @@
+            # if endpoints to keep (unmodified) are not supplied retrieve then from current_enpoints_list
+            # and add them to the list to recreate.
+            for interface in ['public', 'internal', 'admin']:
+                add_endpoint_if_missing(interface=interface,
+                                        endpoints_list=endpoints_list,
+                                        current_endpoint=current_endpoints_list[0])
+            create_endpoints(is_v3=is_v3,
+                             endpoints_list=endpoints_list,
+                             service_id=service_id,
+                             update_mode=True,
+                             region=region,
+                             enabled=enabled)
+
+def create_endpoints(is_v3, endpoints_list, service_id, update_mode=False, region=None, enabled=True):
+    """
+    Create endpoint(s).
+    :param is_v3:  API version.
+    :param endpoints_list: supplied endpoints list.
+    :param service_id: service to attach endpoints.
+    :return: none.
+    """
+    if is_v3:
+        for endpoint_dict in endpoints_list:
+            if not module.check_mode:
+                out = cloud.create_endpoint(service_id,
+                                      url=endpoint_dict.get('url'),
+                                      interface=endpoint_dict.get('interface'),
+                                      region=endpoint_dict.get('region'),
+                                      enabled=endpoint_dict.get('enabled'))
+                if len(out) == 1 and hasattr(out[0], 'id'):
+                    update_info["ids"].append(out[0].id)
+            append_info(is_v3, "creation", endpoint_dict)
+    else:
+        args = {}
+        for endpoint_dict in endpoints_list:
+            args["{}_url".format(endpoint_dict.get('interface'))] = endpoint_dict.get('url')
+        args['region'] = region
+        args['enabled'] = enabled
+        if not module.check_mode:
+            out = cloud.create_endpoint(service_id, **args)
+            if len(out) == 1 and hasattr(out[0], 'id'):
+                update_info["ids"].append(out[0].id)
+        append_info(is_v3, "creation" if update_mode is False else "update", args)
+
+def main():
+
+    global cloud, module, update_info
+
+    update_info = {
+        'ids': [],
+        'list': []
+    }
+
+    argument_spec = openstack_full_argument_spec(
+        service_name=dict(required=True),
+        service_type=dict(required=True),
+        endpoints=dict(required=True, type='list'),
+        region=dict(required=False),
+        enabled=dict(default=True, required=False),
+        state=dict(default='present', choices=['absent', 'present'])
+    )
+
+    module_kwargs = openstack_module_kwargs()
+
+    module = AnsibleModule(argument_spec,
+                           supports_check_mode=True,
+                           **module_kwargs)
+
+    if not HAS_SHADE or (StrictVersion(shade.__version__) < StrictVersion('1.5.0')):
```

Comment:
this should be version 1.11.0

Please follow the messages the other modules do rather than combining them together like this